Python/Product/VSCode/AnalysisVsc/Program.cs

```diff
@@ -51,5 +60,44 @@ internal static class Program {
             }
 #endif
         }
+
+        private sealed class SingleThreadSynchronizationContext : SynchronizationContext, IDisposable {
```

Comment:
From what I understand, `JsonRpc` uses sync context to schedule continuation, so it guarantees the order of method execution start, but there can be any kind of mix on next await.

Reply:
That's correct, and it's what we are aiming for. When reviewing command handlers, please make sure that all critical state (locks/counters/etc.) are updated before the first `await`.